web/src/utils.js

```diff
@@ -159,6 +189,96 @@ export function makeUUID() {
   );
 }
 
+export const icon = {
+  delete: () => {
+    return renderIcon(
+      ['polyline', { points: '3 6 5 6 21 6' }],
+      [
+        'path',
+        {
+          d: 'M19 6v14a2 2 0 0 1-2 2H7a2 2 0 0 1-2-2V6m3 0V4a2 2 0 0 1 2-2h4a2 2 0 0 1 2 2v2',
+        },
+      ],
+      ['line', { x1: 10, x2: 10, y1: 11, y2: 17 }],
+      ['line', { x1: 14, x2: 14, y1: 11, y2: 17 }],
+    );
+  },
+  inspector: () => {
+    const svg = namespace('svg', {
+      fill: 'none',
+      'fill-opacity': 1,
+      height: 26,
+      stroke: 'currentColor',
+      viewBox: '-1 -1 26 26',
+      width: 26,
+    });
+    const g = namespace('g', {});
+    g.append(
+      namespace('circle', {
+        cx: 12,
+        cy: 12,
+        r: 12,
+        'stroke-linecap': 'round',
+        'stroke-linejoin': 'round',
+        'stroke-width': 2,
+      }),
+      namespace('path', {
+        d: 'M 11.88208 4.883789 C 12.326418 4.883789 12.702391 5.039305 13.01001 5.350342 C 13.317628 5.6613785 13.471436 6.035642 13.471436 6.4731445 C 13.471436 6.910647 13.31592 7.283202 13.004883 7.59082 C 12.693846 7.898439 12.319582 8.052246 11.88208 8.052246 C 11.444578 8.052246 11.072023 7.898439 10.764404 7.59082 C 10.456786 7.283202 10.302979 6.910647 10.302979 6.4731445 C 10.302979 6.035642 10.456786 5.6613785 10.764404 5.350342 C 11.072023 5.039305 11.444578 4.883789 11.88208 4.883789 Z M 13.317627 9.528809 L 13.317627 17.126953 C 13.317627 17.803714 13.39624 18.236083 13.553467 18.424072 C 13.710694 18.612061 14.018308 18.719726 14.476318 18.74707 L 14.476318 19.11621 L 9.298096 19.11621 L 9.298096 18.74707 C 9.721926 18.733398 10.036376 18.610353 10.241455 18.37793 C 10.378175 18.220702 10.446533 17.803714 10.446533 17.126953 L 10.446533 11.52832 C 10.446533 10.851559 10.367921 10.41919 10.210693 10.231201 C 10.053466 10.043212 9.74927 9.935547 9.298096 9.908203 L 9.298096 9.528809 Z',
+        fill: 'currentColor',
+      }),
+    );
+    svg.append(g);
+    return svg;
+  },
```

Comment:
This is not as straightforward as icons that uses `renderIcon()` to be stored as a value. Still, it is possible with IIFE:

```js
  inspector: (() => {
    // ...
    return svg;
  })(),
```

Reply:
I don't know if the icon-based up and down symbol will make it into 4.1.0 and if not, we could just do performance testing of 1000 torrents at the meantime in one of the extreme use-cases. I'm sure right now the performance for current icons is going to be same between markup, IIFE, and functions, I'm leaning toward to functions and ideally want to have them make it into 4.1.0. I feel that function is good because it will not spread confusion in element creation. If you're fine with this then please mark this as resolved. Also thanks for the protip about IIFE! I understand the purpose of IIFE now that I'm seeing it sometime in another place.

Reply:
You can go ahead and mark this as resolved, as it seems to be no longer relevant...? (All the JS jargon and hacks are starting to confuse me)

I don't have the "Resolve" button.

Reply:
Yeah you have a point, and okay I didn't realized there's no resolution button for anyone other than the original author.